specs/math.txt

```diff
@@ -71,57 +82,97 @@ $\$$
 $$y = \$ x$$
 .
 <p><span class="math inline">\$</span></p>
-<div class="math block">y = \$ x</div>
+<p><span class="math display">y = \$ x</span></p>
+````````````````````````````````
+
+Inline mode math expressions cannot contain unescaped `$` characters, but
+display mode expressions can, though not two in a row:
+
+```````````````````````````````` example
+$x $ x$
+
+$$ $ $$
+
+$$ $$ $$
+.
+<p>$x $ x$</p>
+<p><span class="math display"> $ </span></p>
+<p><span class="math display"> </span> $$</p>
 ````````````````````````````````
 
-Inline math text with backticks ``$`...`$``:
+Inline math expressions cannot start or end with whitespace, including newlines:
 
 ```````````````````````````````` example
-this is math inline $`y=x`$ test.
-dollar can be contained: $`foo$bar`$
+these are not math texts: $ y=x$, $y=x $, $
+y=x$ and $y=x
+$
 .
-<p>this is math inline <span class="math inline">y=x</span> test.
-dollar can be contained: <span class="math inline">foo$bar</span></p>
+<p>these are not math texts: $ y=x$, $y=x $, $
+y=x$ and $y=x
+$</p>
 ````````````````````````````````
 
-Math expression without backtick cannot start/end with spaces but Math expression with backtick can:
+Inline math expressions do not need to be surrounded with whitespace:
 
 ```````````````````````````````` example
-these are not math texts: $ y=x$ and $y=x $ and $ y=x $
-but these are math texts: $` y=x`$ and $`y=x `$ and $` y=x `$
+these are math texts: foo$y=x$bar and $y=x$bar and foo$y=x$ bar
 .
-<p>these are not math texts: $ y=x$ and $y=x $ and $ y=x $
-but these are math texts: <span class="math inline"> y=x</span> and <span class="math inline">y=x </span> and <span class="math inline"> y=x </span></p>
+<p>these are math texts: foo<span class="math inline">y=x</span>bar and <span class="math inline">y=x</span>bar and foo<span class="math inline">y=x</span> bar</p>
 ````````````````````````````````
 
-Math expression without backtick surrounded with whitespaces:
+Math expressions can be immediately followed by other math expressions:
 
 ```````````````````````````````` example
-these are not math texts: foo$y=x$bar and $y=x$bar and foo$y=x$ bar
-but these are math texts: foo$`y=x`$bar and $`y=x`$bar and foo$`y=x`$ bar
```

Comment:
Can you keep some test cases where alphabetical characters just before/after math expressions?